src/hotspot/share/code/codeBlob.hpp

```diff
@@ -124,13 +122,17 @@ class CodeBlob {
 
   bool _caller_must_gc_arguments;
 
+  address _mutable_data;
+  int     _mutable_data_size;
+
 #ifndef PRODUCT
   AsmRemarks _asm_remarks;
   DbgStrings _dbg_strings;
 #endif
 
   CodeBlob(const char* name, CodeBlobKind kind, CodeBuffer* cb, int size, uint16_t header_size,
-           int16_t frame_complete_offset, int frame_size, OopMapSet* oop_maps, bool caller_must_gc_arguments);
+           int16_t frame_complete_offset, int frame_size, OopMapSet* oop_maps, bool caller_must_gc_arguments,
+           int mutable_data_size = 0);
```

Comment:
If we want to allow the default for mutable data size to be the relocations size, then instead of using = 0 here, you could do this instead:
```
 CodeBlob(const char* name, CodeBlobKind kind, CodeBuffer* cb, int size, uint16_t header_size,
           int16_t frame_complete_offset, int frame_size, OopMapSet* oop_maps, bool caller_must_gc_arguments,
           int mutable_data_size);

 CodeBlob(const char* name, CodeBlobKind kind, CodeBuffer* cb, int size, uint16_t header_size,
           int16_t frame_complete_offset, int frame_size, OopMapSet* oop_maps, bool caller_must_gc_arguments) : 
    CodeBlob(name, kind, cb, size, header_size,
           frame_complete_offset, frame_size, oop_maps, caller_must_gc_arguments,
           cb->total_relocation_size)
{
}
```
but I would prefer not to treat relocations as special, and have the caller always pass the correct value.

Reply:
Agree.